### ISSUE TITLE
Where to download archive data & a suggestion 

### ISSUE THREAD

**Issue by codeser** [alltheplaces/alltheplaces#1555]
I'm noticing many of the files are empty (0 bytes) and from what I understood from @iandees the reason is because the spider ran but didn't produce any data because the website changed most likely. 

But at some point the spider downloaded the data and it's probably still 99% accurate.  I was wondering where I can access an archive of the zip files? 

Or if maybe if the spider fails in the future it shouldn't just erase the file but rather leave the last valid data set but add some sort of flag (or last updated at date) so we know this data is older.

**Comment by caffeinatedMike** [comment]
I'm seeing this issue with about a third of the files from the September file archive as well. This certainly seems like an issue that warrants further investigation.

**To name a few larger ones:**
- Albertsons
- Autozone, 
- Burger King, 
- Costco, 
- Dollar Tree, 
- Kroger

**Comment by Jeremiah-England** [comment]
Yes, a little over a third of them are empty. I've extracted the latest `output..tar.gz` ([download](https://data.alltheplaces.xyz/runs/2020-12-16-14-42-39/output.tar.gz)) to an `output` directory full of geojsons. I get these numbers,

```bash
ls -1sh output/ |  wc -l
# 637
ls -1sh output/ | grep '    0' | wc -l
# 253
```
As you can see, 253 of them are empty currently. 

This does not seem to be due the spider's not working anymore because the website changed. For example, `allstate_insurance_agents.geojson` is one of the empty geojsons but it was recently fixed (https://github.com/alltheplaces/alltheplaces/pull/1663) as @robertaustinbell pointed out to me. I'm running it locally (still running) using the `runspider` command and I'm getting a lot of valid jsons logged in the console. e.g.

``` 
...
'website': 'https://agents.allstate.com/tiffany-soto-ridgewood-ny.html'}
2020-12-19 05:21:54 [scrapy.core.engine] DEBUG: Crawled (200) <GET https://agents.allstate.com/steven-luckert-ridge-ny.html> (referer: https://agents.allstate.com/usa/ny)
2020-12-19 05:21:54 [scrapy.core.scraper] DEBUG: Scraped from <200 https://agents.allstate.com/steven-luckert-ridge-ny.html>
{'addr_full': '1660 Middle Country Rd Ste 2',
 'brand': 'Allstate',
 'brand_wikidata': 'Q2645636',
 'city': 'Ridge',
 'extras': {'@spider': 'allstate_insurance_agents'},
 'lat': 40.893142683657,
 'lon': -72.8970574426421,
 'name': 'Steven Luckert',
 'opening_hours': 'Mo-Tu 09:00-17:00; We 09:00-19:00; Th-Fr 09:00-17:00; Sa '
                  '09:00-12:00',
 'phone': '+16312081200',
 'postcode': '11961',
 'ref': 'steven-luckert-ridge-ny',
 'state': 'NY',
 'website': 'https://agents.allstate.com/steven-luckert-ridge-ny.html'}
2020-12-19 05:21:54 [scrapy.core.engine] DEBUG: Crawled (200) <GET https://agents.allstate.com/usa/ny/richmond-hill> (referer: https://agents.allstate.com/usa/ny)
2020-12-19 05:21:55 [scrapy.core.engine] DEBUG: Crawled (200) <GET https://agents.allstate.com/usa/ny/rhinebeck> (referer: https://agents.allstate.com/usa/ny)
2020-12-19 05:21:56 [scrapy.core.engine] DEBUG: Crawled (200) <GET https://agents.allstate.com/usa/ny/rensselaer> (referer: https://agents.allstate.com/usa/ny)
...
```
I assume this means everything is working well, but I will edit this comment if that changes.

**Comment by Cj-Malone** [comment]
A half fix for this could be keep the outputted data in a git repo so we could go back and find the last good data.

**Comment by iandees** [comment]
Yea, a git archive is interesting. I haven't done it yet because the whole run of spiders takes longer than the maximum time for a Github Action. It would be a lot easier to commit to a repo if I could get it running in a Github Action. I could do something with Github access tokens or the Github API in the current AWS Fargate setup, but I don't have that setup right now.

As for the blank files: I think there's a bug/feature in scrapy where if there's an error in the spider it will truncate the output file and produce an invalid GeoJSON. I'm not sure of a way around that yet.

**Comment by iandees** [comment]
🤖 Both parts of this are resolved now: archive/history access exists via builds.html and spiders.html (see #3698), and the truncated/empty-output problem from spiders killed mid-crawl was fixed by #17554 (`ci/repair_truncated_geojson.py`, wired into `ci/run_all_spiders.sh`), which repairs GeoJSON left incomplete instead of leaving it corrupt or empty.
